modules/cmsis/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory(${ZEPHYR_CURRENT_MODULE_DIR} cmsis)
+zephyr_include_directories(.)
```

Comment:
Adding it like this unconditionally, it gets added to all builds.
So, for simulation, now we depend on the include path order. How things stand right now, this path is after the nrf52_bsim cmsis/ include, which means things did not blow up.
I'd add a ifdef ARM archs here or so, as this headers are the CMSIS Core glue.

Reply:
the include path is unconditional, but same was before with the include found in `zephyr/arch/...`, right? I'm not sure I understand the problem here.

Reply:
That before it was added only if you were building with the ARM archs. Now it is always.

Reply:
As is today, cmsis _glue_ code resides in:

```
include/zephyr/arch/arm/aarch32/cortex_a_r/cmsis.h
include/zephyr/arch/arm/aarch32/cortex_m/cmsis.h
```

Which is always in the include path, no?

Reply:
Only for ARM targets

Reply:
ok, got the problem, bsim has its own `cmsis.h` in the path as well. should now be fixed with `cmsis_core.h` (also, include path is only added if actually makes sense, ie for M/A/R)